dialer/fastconnect.go

```diff
@@ -113,21 +113,21 @@ func (fcd *fastConnectDialer) connectAll(dialers []ProxyDialer) {
 		return
 	}
 	log.Debugf("Dialing all dialers in parallel %#v", dialers)
-outerLoop:
 	for {
+		// Loop until we're connected
+		if len(fcd.connected.dialers) < 2 {
+			fcd.parallelDial(dialers)
+			// Add jitter to avoid thundering herd
+			time.Sleep(time.Duration(rand.Intn(4000)) * time.Millisecond)
+		} else {
+			break
+		}
 		select {
 		case <-fcd.stopCh:
 			log.Debug("Stopping parallel dialing")
 			return
 		default:
-			// Loop until we're connected
-			if len(fcd.connected.dialers) < 2 {
-				fcd.parallelDial(dialers)
-				// Add jitter to avoid thundering herd
-				time.Sleep(time.Duration(rand.Intn(4000)) * time.Millisecond)
-			} else {
-				break outerLoop
-			}
+
 		}
```

Comment:
It's not big deal since the goroutine running `connectAll` would only hang around for a few seconds, at most, but, if you add another `case` and use `time.After` instead of `time.Sleep` here, the goroutine will exit as soon as it's canceled and be cleaned up. Just FYI.

Reply:
Ah good point -- fixed!